internal/backend/remote-state/consul/client.go

```diff
@@ -231,7 +231,11 @@ func (c *RemoteClient) Put(data []byte) error {
 		}
 		// transaction was rolled back
 		if !ok {
-			return fmt.Errorf("consul CAS failed with transaction errors: %v", resp.Errors)
+			var resultErr error
+			for _, respError := range resp.Errors {
+				resultErr = append(resultErr, errors.New(respError.What))
```

Comment:
Oops my bad, I made a mistake in my original suggestion :facepalm:

This should be:

```suggestion
				resultErr = multierror.Append(resultErr, errors.New(respError.What))
```